cicecore/cicedynB/dynamics/ice_dyn_shared.F90

```diff
@@ -916,15 +922,197 @@ subroutine basal_stress_coeff (nx_block, ny_block,         &
+      real (kind=dbl_kind) :: atot, x_kmax, x997
+      real (kind=dbl_kind) :: cut, rhoi, rhow, gravit, pi, puny
+
+      character(len=*), parameter :: subname = '(seabed2_stress_coeff)'
+
+      call icepack_query_parameters(rhow_out=rhow, rhoi_out=rhoi)
+      call icepack_query_parameters(gravit_out=gravit)
+      call icepack_query_parameters(pi_out=pi)    
+      call icepack_query_parameters(puny_out=puny)
+
+      Tbt=c0
+      sigma_b = 2.5d0 ! Standard deviation of bathymetry. 
 
+      do ij = 1, icellt
+         i = indxti(ij)
+         j = indxtj(ij)
+
+         atot = sum(aicen(i,j,1:ncat))
+
+         if (atot .gt. 0.05_dbl_kind .and. hwater(i,j) .lt. max_depth) then
+
+            mu_b = hwater(i,j)           ! mean of PDF (normal dist) bathymetry
+            wid_i = max_depth/ncat_i     ! width of ice categories
+            wid_b = c2*c3*sigma_b/ncat_b ! width of bathymetry categories
+
+            x_k = (/(wid_i*(i*c1-0.5d0), i=1, ncat_i)/) 
+            y_n = (/( (mu_b-c3*sigma_b)+(i*c1-0.5d0)*(c2*c3*sigma_b/ncat_b), i=1, ncat_b )/)
+
+            vcat(1:ncat) = vicen(i,j,1:ncat)
+            acat(1:ncat) = aicen(i,j,1:ncat)
+
+            m_i = sum(vcat)
+
+            v_i=c0
+            do n =1, ncat
+               v_i = v_i + vcat(n)**2 / (max(acat(n), puny))
+            enddo
+            v_i = v_i - m_i**2
+
+            mu_i    = log(m_i/sqrt(c1 + v_i/m_i**2)) ! parameters for the log-normal
+            sigma_i = sqrt(log(c1 + v_i/m_i**2))
+
+            ! max thickness associated with percentile of log-normal PDF
+            ! x997 was obtained from an optimization procedure (Dupont et al.)  
+
+!            x95  = exp(mu_i + sqrt(c2*sigma_i)*1.1631d0)
+!            x99  = exp(mu_i + sqrt(c2*sigma_i)*1.6450d0)
+!            x995 = exp(mu_i + sqrt(c2*sigma_i)*1.8214d0)
+!            x996 = exp(mu_i + sqrt(c2*sigma_i)*1.8753d0)
+            x997 = exp(mu_i + sqrt(c2*sigma_i)*1.9430d0)
+!            x998 = exp(mu_i + sqrt(c2*sigma_i)*2.0352d0)
+!            x999 = exp(mu_i + sqrt(c2*sigma_i)*2.1851d0)
+!            xinf = exp(mu_i + sqrt(c2*sigma_i)*4.4981d0) ! 0.9999999999
+            x_kmax = x997
+
+            ! Set x_kmax to hlev of the last category where there is ice
+            ! when there is no ice in the last category           
+            cut = x_k(ncat_i)
+            do n = ncat,-1,1
+               if (acat(n) < puny) then
+                  cut = hin_max(n-1)
+               else
+                  exit
+               endif
+            enddo
+            x_kmax = min(cut, x_kmax)
+
+            g_k = exp(-(log(x_k) - mu_i) ** 2 / (c2 * sigma_i ** 2)) / (x_k * sigma_i * sqrt(c2 * pi))
+
+            b_n  = exp(-(y_n - mu_b) ** 2 / (c2 * sigma_b ** 2)) / (sigma_b * sqrt(c2 * pi))
+
+            P_x = g_k*wid_i
+            P_y = b_n*wid_b
+
+            do n =1, ncat_i
+               if (x_k(n) .gt. x_kmax) P_x(n)=c0
+            enddo
```

Comment:
minor: I think maybe this loop could be rewritten with  `where `statement

Reply:
and > instead of .gt.

and below, < for .le., == for .eq.